firestore-bigquery-export/firestore-bigquery-change-tracker/src/bigquery/index.ts

```diff
@@ -191,17 +192,29 @@ export class FirestoreBigQueryEventHistoryTracker
    * A half a second delay is added per check while the function
    * continually re-checks until the referenced dataset and table become available.
    */
-  private async waitForInitialization(dataset, table) {
+  private async waitForInitialization() {
     return new Promise((resolve) => {
       let handle = setInterval(async () => {
-        const [datasetExists] = await dataset.exists();
-        const [tableExists] = await table.exists();
+        try {
+          const dataset = this.bigqueryDataset();
+          const changelogName = this.rawChangeLogTableName();
+          const table = dataset.table(changelogName);
+
+          const [datasetExists] = await dataset.exists();
+          const [tableExists] = await table.exists();
+
+          if (datasetExists && tableExists) {
+            clearInterval(handle);
+            return resolve(table);
+          }
+        }
 
-        if (datasetExists && tableExists) {
+        catch (ex) {
           clearInterval(handle);
-          return resolve(table);
+          logs.failedToInitializeWait(ex.message);
         }
-      }, 500);
+
+      }, 5000);
```

Comment:
Small nit; before it was 500, now 5000 - is there any side effect we need to mention here? e.g. documentation

Reply:
The original `5000` was introduced to give enough time should the `tables` need enough to create.

This seemed quite lengthy and was reduced with testing for performance (time to taken to receive writes to the database). There is no side effect, but now the PR has been completed. Perhaps `5000` should be reintroduced to limit the chance of writing before a table has been created.

`5000` certainly seems safer than `500`.

Reply:
does this mean any write firestore will take a minimum of 5 seconds to propagate to bigquery in the future?

Reply:
@meyerovb This would only occur if an error was found with the BigQuery api when writing the record. 

The errors typically appear when a large amount of writes happen simultaneously.